### PR TITLE
Fix function name

### PR DESCRIPTION
**Pull request by Xunzhuo**
# Description

fix function name

This bug fixes SetExtenstionProtocolOptions( ) to SetExtensionProtocolOptions( )

# Context

Users ran into this bug doing ... \ Users needed this feature to ...

# Checklist:

- [ ] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [ ] If I updated APIs (our protos) or helm values, I ran `make install-go-tools generated-code` to ensure there will be no code diff
- [ ] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [ ] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/4786

### CODE REVIEW

**Comment by kdorosh** [comment]
Hi @Xunzhuo please add a changelog to your PR per https://github.com/solo-io/go-utils/tree/master/changelogutils#changelog to pass that check, and ensure that our codegen check will pass as well. Thanks for the contribution!

**Comment by Xunzhuo** [comment]
resolve #4786

**Comment by Xunzhuo** [comment]
@kdorosh Done. Please check it

File: changelog/v1.8.0-beta18/standard-function.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+changelog:
+  - type: FIX
```
Comment (kdorosh) [inline changelog/v1.8.0-beta18/standard-function.yaml:2]: we should change this to `NON_USER_FACING`

**Comment by kdorosh** [comment]
/kick different aws test flake https://console.cloud.google.com/cloud-build/builds;region=global/d84b9df8-ed77-464a-9988-b160dd7b73c4;step=10?project=solo-public

**Comment by Xunzhuo** [comment]
@kdorosh Changelog type has fixed, maybe we should test again

**Review by kdorosh — APPROVED** [review]

**Review by sam-heilbron — APPROVED** [review]

**Pull request merged**